### ISSUE TITLE
my Coolprop in EES does not work

### ISSUE THREAD

**Issue by omidne** [CoolProp/CoolProp#1581]
Hello 

I put EES library of coolpropsi in the mentioned folder but it doesnt work. 

my coolprop cant recognize the parameter I think!

it wrote that: """Your input key #1 [p] is not valid. (names are case sensitive)"

**Comment by omidne** [comment]
As a simple call I used this 

fluid$='R134a'
P_ex_cd=5e5 [Pa]
DELTAT_ex_cd = coolpropsi('T',fluid$,'X',1,'P',P_ex_cd)

but itsnot working

**Comment by davideziviani** [comment]
Have you tried to run the provided example? The syntax does not look right to me...

" Calculate the normal boiling point temperature of water in K"
fluid$ = 'Water'
Tnbp = PropsSI('T','P',101325,'Q',1,fluid$)
 
 " Calculate the density of a 50/50 molar blend of methane and ethane at STP "
xx = string$(0.5)
yy = string$(0.5)
a$='HEOS::Methane['||yy||']&Ethane['||xx||']'
D_STP = PropsSI('D', 'P', 101325, 'T', 298.15, a$)

**Comment by ibell** [comment]
Yes, when in doubt, first try to follow exactly the provided examples.  In
this case, though I can't verify this, I believe there is also the
coolpropsi function in the EES wrapper because some of my colleagues wanted
me to add that function.  But definitely start with the PropsSI function.

**Comment by davideziviani** [comment]
@ibell : I have opened up the EES wrapper and the coolpropsi function accepts only lower case inputs otherwise it throws an error. That explains why DELTAT_ex_cd = coolpropsi('T',fluid$,'X',1,'P',P_ex_cd) does not work. See below:

Function coolpropsi(output$,fluid$,input1$,input1val,input2$,input2val)
 
"Format the first input with the right synthax and units for CoolProp:"
If lowercase$(input1$)='t' Then
   input1val = input1val + 273.15
   in1$='T'
Else
    If lowercase$(input1$)='p' Then
        input1val = input1val/ 1000
        in1$='P'
   Else
        If lowercase$(input1$)='rho' Then
            in1$='D'
        Else
            If lowercase$(input1$)='h' Then
                input1val = input1val/ 1000
                in1$='H'
            Else
                    If lowercase$(input1$)='s' Then
                          input1val = input1val/1000
                          in1$='S'
                    Else
                           If lowercase$(input1$)='u' Then
                                input1val = input1val/ 1000
                                in1$='U'
                            Else
                                    If lowercase$(input1$)='x' Then
                                        in1$='Q'
                                   Else
                                        Call error('Invalid input key1: XXX$',input1$)
                                   Endif
                            Endif
                     Endif
               Endif
        Endif
    Endif
Endif

**Comment by ibell** [comment]
Actually, that just means that first it takes the lower case and then
compares it to 'p', so that would match either 'P' or 'p'.  I think the
problem with that example is something else...

**Comment by ibell** [comment]
Closing as part of an issue-tracker cleanup — this issue has had no activity in over two years.

*To be transparent: this comment and the close action are being performed by Claude (Anthropic's AI coding assistant), running on behalf of @ibell. Ian reviewed and approved the bulk-close criteria; I am the one executing the action.*

If you would like this issue reconsidered, please feel free to **re-open it** — a fresh comment with current behaviour against the latest CoolProp release is the most useful way to revive the discussion.

— Claude (acting for @ibell)
